libs/vkd3d/command.c

```diff
@@ -7433,13 +7513,25 @@ static void STDMETHODCALLTYPE d3d12_command_list_ResolveQueryData(d3d12_command_
             return;
         }
 
-        copy_region.srcOffset = stride * start_index;
-        copy_region.dstOffset = buffer->heap_offset + aligned_dst_buffer_offset;
-        copy_region.size = stride * query_count;
+        if (type != D3D12_QUERY_TYPE_BINARY_OCCLUSION)
+        {
+            copy_region.srcOffset = stride * start_index;
+            copy_region.dstOffset = buffer->heap_offset + aligned_dst_buffer_offset;
+            copy_region.size = stride * query_count;
 
-        VK_CALL(vkCmdCopyBuffer(list->vk_command_buffer,
-                query_heap->vk_buffer, buffer->vk_buffer,
-                1, &copy_region));
+            VK_CALL(vkCmdCopyBuffer(list->vk_command_buffer,
+                    query_heap->vk_buffer, buffer->vk_buffer,
+                    1, &copy_region));
+        }
+        else
+        {
+            uint32_t dst_index = aligned_dst_buffer_offset / sizeof(uint64_t);
+
+            d3d12_command_list_resolve_binary_occlusion_queries(list,
```

Comment:
Shouldn't this be handled in the resolve shader with a spec const bool? I figure it would just write 1 or 0 depending on the final counter value.

Reply:
We can't do that since you can allocate both binary and non-binary queries from an occlusion query heap.

There's the option to do this per query during the resolve by passing extra data to the shader, but there's a lot of room for the app to fuck up by doing this:
```
cmd->BeginQuery(..., D3D11_QUERY_TYPE_OCCLUSION);
...
cmd->EndQuery(..., D3D11_QUERY_TYPE_OCCLUSION);
cmd->ResolveQueryData(..., D3D11_QUERY_TYPE_BINARY_OCCLUSION);
```
in which case we should still do a binary resolve, even if it was specified incorrectly for the queries.

Reply:
Ah, ok. I didn't realize app could resolve with BINARY separately.